0001-basic-expiry-model.md

```diff
@@ -0,0 +1,166 @@
+# Basic model to calculate expiry times
+
+* Status: proposed
+* Authors: [Lloyd Fournier](@LLFourn), [Lucas Soriano](@luckysori)
+* Date: 2019-03-05
+
+Tracking issue: [#14](https://github.com/comit-network/RFCs/issues/14)
+
+## Context
+
+The protocol defined in [RFC003](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-basic.md) introduces the concept of expiry times, which mark the time when assets can be refunded to their original owners.
+Values that are reasonable and that ensure that the protocol is secure for both parties must be estimated.
+
+## Research
+
+### Findings
+
+Here are the results of initial discussions:
+
+- The difference between `alpha_expiry` and `beta_expiry` has an effect on the security of the model for *Bob*.
+This does not affect Alice since she will have already redeemed `beta_asset` by the time this is relevant.
+- `beta_expiry` has an effect on the time that *Alice* will have to redeem `beta_asset` and therefore on the likelihood that the protocol will be completed successfully.
+This does _not_ have any direct effect on the security of the protocol, but rather on its *execution guarantees*.
+
+Given this, the topic of this spike changed from creating a security model to, more generally, creating a basic model to calculate expiry times.
+
+### Alpha expiry
+
+There are two ways in which `alpha_expiry`, <img src="https://latex.codecogs.com/gif.latex?E_{\alpha}"/>, has an effect on the protocol:
+
+1. How long Bob will (at least) have to redeem `alpha_asset` if Alice redeems `beta_asset`.
+2. How long Alice will have to wait to be able to refund `alpha_asset` if the protocol isn't carried out to completion.
+
+It would therefore be ideal to find a value of `alpha_expiry` that lets Bob redeem `alpha_asset` with high probability while also ensuring Alice doesn't have to wait too long to be able to be refunded if it comes to that.
+
+#### Assumptions
+
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?T_{\alpha}"/>, the time in which he is overwhelmingly confident he will be able to confirm a transaction on `alpha_ledger`.
+For simplicity, the model assumes that the probability of this happening within the time provided is 1.
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations he would need to have on his redeem transaction before he considers `alpha_asset` to be his own.
+- Block time on `alpha_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda_{\alpha}"/>.
+- A confirmation is given by the arrival of a new block.
```

Comment:
Is the chance of finding a block in particular time important? What if there are no blocks?

Reply:
Please link to erlang and exponential distribution.